FreeType/Support/Advance.hsc

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
Do you need TemplateHaskell here (this applies to some other modules on second glance, too)?

Reply:
Pretty much all modules that throw `ftError`s require `TemplateHaskell`, because I pass the `Name` of the function instead of writing it as a `String` (so that I don't mistype a function name).

The `template-haskell` dependency is required by `microlens-th` already, so the only negative is a slightly longer compilation time in my eyes. However if the `microlens-th` is to be moved into a separate `freetype2-lens` package, then I guess I can just regex them into `String`s.